CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_C_STANDARD 11)
 option(BUILD_ENTERPRISE "Set whether or not to build enterprise edition" OFF)
 option(CODE_COVERAGE_ENABLED "Set whether or not code coverage report should be generated" OFF)
+option(DISABLE_LTO_BUILD "Disable build with Link-time optimization" OFF)
```

Comment:
This is being added without any functionality, so why?  I don't think it needs to be here just to make it available.  You can still set it and have Core react to it.

Reply:
LTO itself is only applicable to production builds, and was being disabled for tests only

Reply:
Clang has LTO off by default and after the our sync, indeed this line is redundant so I'll close this PR